src/index.ts

```diff
@@ -1,6 +1,9 @@
+import { configureServer } from '@core/db/config.js';
 import fastify from 'fastify';
 import { userRoutes } from './api/users/users.routes.js';
 
+configureServer();
```

Comment:
```suggestion
await configureServer();
```

Acho que é importante aguardar se deu certo a conexão com o banco porque senão nem vale a pena subir o server.